ramips: add support for RAK633

eikepedia opened the pull request:
Specification:

CPU: MT7628 580 MHz. MIPS 24K
RAM: 64 MB
Flash: 8 MB
USB 2.0
5 Port ethernet switch

Product Specification
http://docs.rakwireless.com/en/RAK633/Hardware%20Design/RAK633_Product_Specification_V1.0.pdf

Installation:
Use the Installed uboot Bootloader. Connect a cable to serialport 0. Turn power on.
Choose the option: “Load system code then write to Flash via TFTP” and follow the instructions.

Notes:
The I2C Kernel module work not correctly. You can send and receive data. But the command i2cdetect doesn’t work. FS#845

Signed-off-by: Eike Feldmann eike.feldmann@outlook.com

eikepedia committed:
rampis: add support für Rakwireless RAK633

Specification:

CPU: 		MT7628 580 MHz. MIPS 24K
RAM:		64 MB
Flash:		8 MB
USB: 		1x Port USB 2.0
Ethernet:	5 Port ethernet switch
WIFI: 		802.11n/g/b 20/40 MHz
UART:		2x

Product Specification
http://docs.rakwireless.com/en/RAK633/Hardware%20Design/RAK633_Product_Specification_V1.0.pdf

Installation:
Use the installed uboot Bootloader. Connect a serial cable to serialport 0. Turn power on.
Choose the option: “Load system code then write to Flash via TFTP”.
Choose the local devive IP and the TFTP server IP and the file name of the system image.
After if the Bootloader will copy the image to the local flash.

Notes:
The I2C Kernel module work not correctly. You can send and receive data. But the command i2cdetect doesn’t work. FS#845

Signed-off-by: Eike Feldmann <eike.feldmann@outlook.com>

ynezz commented on target/linux/ramips/dts/RAK633.dts:
```diff
@@ -0,0 +1,97 @@
+/dts-v1/;
```
@eikepedia  Please can you consider adding `SPDX-License-Identifier: GPL-2.0-or-later OR MIT` as in [DTS Checklist](https://openwrt.org/submitting-patches#dts_checklist) ?
> Having MIT as alternative is sometimes preferred by upstream maintainers
and allows sharing that simple code with other projects. We don't really
want multiple DTS versions for the same device.

ynezz commented on target/linux/ramips/dts/RAK633.dts:
```diff
@@ -0,0 +1,97 @@
+
+/ {
+	compatible = "rakwireless,rak633", "mediatek,mt7628an-soc";
+	model = "RAKwireless RAK633";
```
@eikepedia Please add new line here.

ynezz commented on target/linux/ramips/dts/RAK633.dts:
```diff
@@ -0,0 +1,97 @@
+	
+	leds {
+		compatible = "gpio-leds";
+			status {
```
@eikepedia Looks like you've some whitespace/indent issue here.

ynezz commented on target/linux/ramips/dts/RAK633.dts:
```diff
@@ -0,0 +1,97 @@
+			status {
+				label = "rak633:blue:status";
+				gpios = <&gpio1 12 GPIO_ACTIVE_LOW>;
+				};
```
@eikepedia and here

ynezz commented on target/linux/ramips/dts/RAK633.dts:
```diff
@@ -0,0 +1,97 @@
+&spi0 {
+	status = "okay";
+
+	m25p80@0 {
```
@eikepedia Could you please change this to generic name `flash` as suggested by [DTS Checklist](https://openwrt.org/submitting-patches#dts_checklist) and check if it still boots?
>  5. The name of a node should reflect the function of the device and not its model. Examples for generic node names can be found in Section 2.2.2 Generic Names Recommendation

ynezz commented on target/linux/ramips/image/mt76x8.mk:
```diff
@@ -156,6 +156,13 @@ define Device/pbr-d1
 
+define Device/rakwireless_rak633
+  DTS := RAK633
+  DEVICE_TITLE := RAKwireless RAK633
```
> Shenzhen Rakwireless Technology Co., Ltd. (RAK) was established in June 2014. 

@eikepedia So it should be probably written as `Rakwireless` and not RAKwireless.

ynezz commented on target/linux/ramips/base-files/etc/board.d/01_leds:
```diff
@@ -279,6 +279,9 @@ netgear,r6350)
 	ucidef_set_led_netdev "wan" "wan" "$boardname:green:wan" eth0.2
 	set_wifi_led "$boardname:green:wifi"
 	;;
+rak633)
```
@eikepedia Please use board name `rakwireless,rak633` from DTS compatible

ynezz commented on target/linux/ramips/base-files/etc/board.d/02_network:
```diff
@@ -115,6 +115,10 @@ ramips_setup_interfaces()
 	psg1208|\
 	psg1218a|\
 	r6220|\
+	rak633)
```
@eikepedia Please use board name `rakwireless,rak633` from DTS compatible

ynezz commented on target/linux/ramips/base-files/lib/ramips.sh:
```diff
@@ -415,6 +415,9 @@ ramips_board_detect() {
 	*"R6220")
 		name="r6220"
 		;;
+	*"RAKwireless RAK633")
```
@eikepedia You can drop this as it's preferred to use board name from DTS compatible.

ynezz commented:
@eikepedia The PR has formal issues as the commit has no `Signed-off-by`, it has wrong commit subject (should probably be `ramips: Add support for Rakwireless RAK633`), missing commit message (hardware info, flashing instructions), please check [Submitting patches](https://openwrt.org/submitting-patches#submitting_patches) for details.

Please don't close this PR and open a new one. Please use `git commit --amend` to add changes to your commit. Use `git push --force` to overwrite your github repository/branch. The PR will afterwards update automatically. See the [squashing commits instructions](https://openwrt.org/submitting-patches#squashing_commits) on how to modify this PR without closing it and open a new one.

ynezz commented:
@eikepedia Pulled into my staging tree at https://git.openwrt.org/openwrt/staging/ynezz.git but I had to make quite a lot of changes, as I've discovered them once I've merged it to my tree locally, as I didn't wanted to waste more time on this, I've made those changes myself. So please check it again and let me know if it's OK with you and you agree with the changes I've made to this PR.

BTW I'm wondering if you could post a hexdump of the factory partition `hexdump -n 256 -C /dev/mtd2`, so we could add MAC address to the WLAN.

eikepedia commented:
@ynezz The changes are ok and I agree

eikepedia commented:
The Output from hexdump -n 256 -C /dev/mtd2

root@OpenWrt:/# hexdump -n 256 -C /dev/mtd2
00000000  28 76 00 02 e0 b9 4d f9  90 76 00 00 00 00 00 00  |(v....M..v......|
00000010  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000020  00 00 00 00 20 00 00 00  e0 b9 4d f9 90 76 e0 b9  |.... .....M..v..|
00000030  4d f9 90 77 22 34 00 20  ff ff 00 01 00 00 00 00  |M..w"4. ........|
00000040  00 00 22 00 00 00 00 00  30 00 00 00 00 00 00 00  |..".....0.......|
00000050  82 00 00 94 40 b0 c0 ca  22 81 81 81 40 ca 22 80  |....@..."...@.".|
00000060  80 81 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000070  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
*
000000a0  c6 c6 c4 c4 c4 c0 c0 c4  c4 c4 c4 c4 c0 c0 00 00  |................|
000000b0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000000c0  00 00 00 00 00 00 00 00  00 1a 22 2a 31 35 01 35  |.........."*15.5|
000000d0  39 40 46 4d 7f 7f 7f 00  00 00 00 00 00 00 00 00  |9@FM............|
000000e0  11 1d 11 1d 1c 35 1c 35  1e 35 1e 35 17 19 17 19  |.....5.5.5.5....|
000000f0  02 00 00 00 d3 80 80 88  0a 00 00 00 00 00 00 00  |................|
00000100

ynezz commented:
> 00000000 28 76 00 02 e0 b9 4d f9 90 76

It looks like a valid MAC e0:b9:4d:f9:90:76 at offset 4, so I've added this MAC to wmac node  https://git.openwrt.org/?p=openwrt/staging/ynezz.git;a=commit;h=426ddbe7ea44e39c71cf73611ca0c4ae32e9a279 please can you compile it and provide me with output of `dmesg` and of `ip address` commands?

eikepedia commented:
I have build the commit.
The Output from _ip address_
root@OpenWrt:/# ip address
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UNKNOWN qlen 1000
    link/ether 2a:cb:a9:1e:83:42 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::28cb:a9ff:fe1e:8342/64 scope link 
       valid_lft forever preferred_lft forever
3: wlan0: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether e0:b9:4d:f9:90:76 brd ff:ff:ff:ff:ff:ff
4: br-lan: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP qlen 1000
    link/ether 2a:cb:a9:1e:83:42 brd ff:ff:ff:ff:ff:ff
    inet 192.168.1.1/24 brd 192.168.1.255 scope global br-lan
       valid_lft forever preferred_lft forever
    inet6 fdb0:ff76:c1ea::1/60 scope global 
       valid_lft forever preferred_lft forever
    inet6 fe80::28cb:a9ff:fe1e:8342/64 scope link 
       valid_lft forever preferred_lft forever
5: eth0.2@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UP qlen 1000
    link/ether 2a:cb:a9:1e:83:42 brd ff:ff:ff:ff:ff:ff
6: eth0.1@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP qlen 1000
    link/ether 2a:cb:a9:1e:83:43 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::28cb:a9ff:fe1e:8343/64 scope link 
       valid_lft forever preferred_lft forever

The output from _dmesg_
root@OpenWrt:/# dmesg
[    0.000000] Linux version 4.14.111 (eike@linux-x00x) (gcc version 7.4.0 (OpenWrt GCC 7.4.0 r9846-d2cd841612)) #0 Mon Apr 8 19:30:42 2019
[    0.000000] Board has DDR2
[    0.000000] Analog PMU set to hw control
[    0.000000] Digital PMU set to hw control
[    0.000000] SoC Type: MediaTek MT7628AN ver:1 eco:2
[    0.000000] bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019655 (MIPS 24KEc)
[    0.000000] MIPS: machine is Rakwireless RAK633
[    0.000000] Determined physical RAM map:
[    0.000000]  memory: 04000000 @ 00000000 (usable)
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] On node 0 totalpages: 16384
[    0.000000] free_area_init_node: node 0, pgdat 80499150, node_mem_map 81000040
[    0.000000]   Normal zone: 128 pages used for memmap
[    0.000000]   Normal zone: 0 pages reserved
[    0.000000]   Normal zone: 16384 pages, LIFO batch:3
[    0.000000] random: get_random_bytes called from start_kernel+0x8c/0x474 with crng_init=0
[    0.000000] pcpu-alloc: s0 r0 d32768 u32768 alloc=1*32768
[    0.000000] pcpu-alloc: [0] 0 
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 16256
[    0.000000] Kernel command line: console=ttyS0,57600 rootfstype=squashfs,jffs2
[    0.000000] PID hash table entries: 256 (order: -2, 1024 bytes)
[    0.000000] Dentry cache hash table entries: 8192 (order: 3, 32768 bytes)
[    0.000000] Inode-cache hash table entries: 4096 (order: 2, 16384 bytes)
[    0.000000] Writing ErrCtl register=0002d9d0
[    0.000000] Readback ErrCtl register=0002d9d0
[    0.000000] Memory: 59724K/65536K available (3677K kernel code, 175K rwdata, 860K rodata, 208K init, 200K bss, 5812K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS: 256
[    0.000000] intc: using register map from devicetree
[    0.000000] CPU Clock: 575MHz
[    0.000000] timer_probe: no matching timers found
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 6647862422 ns
[    0.000012] sched_clock: 32 bits at 287MHz, resolution 3ns, wraps every 7469508094ns
[    0.015369] Calibrating delay loop... 380.92 BogoMIPS (lpj=1904640)
[    0.087545] pid_max: default: 32768 minimum: 301
[    0.096897] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.109855] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.129331] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.148710] futex hash table entries: 256 (order: -1, 3072 bytes)
[    0.160811] pinctrl core: initialized pinctrl subsystem
[    0.172106] NET: Registered protocol family 16
[    0.207342] mt7621_gpio 10000600.gpio: registering 32 gpios
[    0.218570] mt7621_gpio 10000600.gpio: registering 32 gpios
[    0.229693] mt7621_gpio 10000600.gpio: registering 32 gpios
[    0.245239] clocksource: Switched to clocksource MIPS
[    0.256363] NET: Registered protocol family 2
[    0.265755] TCP established hash table entries: 1024 (order: 0, 4096 bytes)
[    0.279455] TCP bind hash table entries: 1024 (order: 0, 4096 bytes)
[    0.291977] TCP: Hash tables configured (established 1024 bind 1024)
[    0.304683] UDP hash table entries: 256 (order: 0, 4096 bytes)
[    0.316145] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes)
[    0.328734] NET: Registered protocol family 1
[    0.337262] PCI: CLS 0 bytes, default 32
[    0.340611] Crashlog allocated RAM at address 0x3f00000
[    0.352480] workingset: timestamp_bits=30 max_order=14 bucket_order=0
[    0.370901] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.382342] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.411767] io scheduler noop registered
[    0.419443] io scheduler deadline registered (default)
[    0.430367] Serial: 8250/16550 driver, 3 ports, IRQ sharing disabled
[    0.444106] console [ttyS0] disabled
[    0.451134] 10000c00.uartlite: ttyS0 at MMIO 0x10000c00 (irq = 28, base_baud = 2500000) is a 16550A
[    0.468963] console [ttyS0] enabled
[    0.482692] bootconsole [early0] disabled
[    0.499153] 10000d00.uart1: ttyS1 at MMIO 0x10000d00 (irq = 29, base_baud = 2500000) is a 16550A
[    0.517246] 10000e00.uart2: ttyS2 at MMIO 0x10000e00 (irq = 30, base_baud = 2500000) is a 16550A
[    0.535207] cacheinfo: Failed to find cpu0 device node
[    0.545420] cacheinfo: Unable to detect cache hierarchy for CPU 0
[    0.558263] spi-mt7621 10000b00.spi: sys_freq: 191666666
[    0.585818] m25p80 spi0.0: s25fl064k (8192 Kbytes)
[    0.595370] 4 fixed-partitions partitions found on MTD device spi0.0
[    0.607960] Creating 4 MTD partitions on "spi0.0":
[    0.617464] 0x000000000000-0x000000030000 : "u-boot"
[    0.628241] 0x000000030000-0x000000040000 : "u-boot-env"
[    0.639682] 0x000000040000-0x000000050000 : "factory"
[    0.650611] 0x000000050000-0x000000800000 : "firmware"
[    0.664476] 2 uimage-fw partitions found on MTD device firmware
[    0.676270] Creating 2 MTD partitions on "firmware":
[    0.686119] 0x000000000000-0x000000181d49 : "kernel"
[    0.696907] 0x000000181d49-0x0000007b0000 : "rootfs"
[    0.707636] mtd: device 5 (rootfs) set to be root filesystem
[    0.720417] 1 squashfs-split partitions found on MTD device rootfs
[    0.732729] 0x0000003d0000-0x0000007b0000 : "rootfs_data"
[    0.745074] libphy: Fixed MDIO Bus: probed
[    0.764037] rt3050-esw 10110000.esw: link changed 0x00
[    0.775110] mtk_soc_eth 10100000.ethernet: generated random MAC address 2a:cb:a9:1e:83:42
[    0.792075] mtk_soc_eth 10100000.ethernet eth0: mediatek frame engine at 0xb0100000, irq 5
[    0.810243] NET: Registered protocol family 10
[    0.822928] Segment Routing with IPv6
[    0.830361] NET: Registered protocol family 17
[    0.839222] 8021q: 802.1Q VLAN Support v1.8
[    0.858035] VFS: Mounted root (squashfs filesystem) readonly on device 31:5.
[    0.873196] Freeing unused kernel memory: 208K
[    0.882019] This architecture does not have kernel memory protection.
[    2.026224] init: Console is alive
[    2.033209] init: - watchdog -
[    2.475269] random: fast init done
[    3.266040] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    3.362100] usbcore: registered new interface driver usbfs
[    3.373145] usbcore: registered new interface driver hub
[    3.383801] usbcore: registered new device driver usb
[    3.399495] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
[    3.414185] ehci-platform: EHCI generic platform driver
[    3.434874] phy phy-10120000.usbphy.0: remote usb device wakeup disabled
[    3.448156] phy phy-10120000.usbphy.0: UTMI 16bit 30MHz
[    3.458524] ehci-platform 101c0000.ehci: EHCI Host Controller
[    3.469946] ehci-platform 101c0000.ehci: new USB bus registered, assigned bus number 1
[    3.485766] ehci-platform 101c0000.ehci: irq 26, io mem 0x101c0000
[    3.525259] ehci-platform 101c0000.ehci: USB 2.0 started, EHCI 1.00
[    3.538696] hub 1-0:1.0: USB hub found
[    3.546573] hub 1-0:1.0: 1 port detected
[    3.557779] ohci_hcd: USB 1.1 'Open' Host Controller (OHCI) Driver
[    3.571751] ohci-platform: OHCI generic platform driver
[    3.582459] ohci-platform 101c1000.ohci: Generic Platform OHCI controller
[    3.595966] ohci-platform 101c1000.ohci: new USB bus registered, assigned bus number 2
[    3.611745] ohci-platform 101c1000.ohci: irq 26, io mem 0x101c1000
[    3.690192] hub 2-0:1.0: USB hub found
[    3.698123] hub 2-0:1.0: 1 port detected
[    3.709768] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    3.734628] init: - preinit -
[    4.436752] usb 1-1: new high-speed USB device number 2 using ehci-platform
[    5.117822] rt3050-esw 10110000.esw: link changed 0x00
[    5.310357] random: procd: uninitialized urandom read (4 bytes read)
[    8.427785] mount_root: jffs2 not ready yet, using temporary tmpfs overlay
[    8.474904] urandom-seed: Seed file not found (/etc/urandom.seed)
[    8.615095] procd: - early -
[    8.621725] procd: - watchdog -
[    9.349411] procd: - watchdog -
[    9.356013] procd: - ubus -
[    9.535689] random: ubusd: uninitialized urandom read (4 bytes read)
[    9.621416] random: ubusd: uninitialized urandom read (4 bytes read)
[    9.634637] random: ubusd: uninitialized urandom read (4 bytes read)
[    9.648249] procd: - init -
[   10.011607] kmodloader: loading kernel modules from /etc/modules.d/*
[   10.032064] ip6_tables: (C) 2000-2006 Netfilter Core Team
[   10.052902] Loading modules backported from Linux version v4.19.32-0-g3a2156c839c7
[   10.067970] Backport generated by backports.git v4.19.32-1-0-g1c4f7569
[   10.084359] ip_tables: (C) 2000-2006 Netfilter Core Team
[   10.104539] nf_conntrack version 0.5.0 (1024 buckets, 4096 max)
[   10.181593] xt_time: kernel timezone is -0000
[   10.261280] mt76_wmac 10300000.wmac: ASIC revision: 76280001
[   10.811774] rt3050-esw 10110000.esw: link changed 0x01
[   11.287448] mt76_wmac 10300000.wmac: Firmware Version: 20151201
[   11.299242] mt76_wmac 10300000.wmac: Build Time: 20151201183641
[   11.325251] mt76_wmac 10300000.wmac: firmware init done
[   11.503136] ieee80211 phy0: Selected rate control algorithm 'minstrel_ht'
[   11.528340] PPP generic driver version 2.4.2
[   11.543808] NET: Registered protocol family 24
[   11.570403] kmodloader: done loading kernel modules from /etc/modules.d/*
[   12.786015] urandom_read: 5 callbacks suppressed
[   12.786025] random: jshn: uninitialized urandom read (4 bytes read)
[   39.904652] rt3050-esw 10110000.esw: link changed 0x00
[   45.405319] jffs2_scan_eraseblock(): End of filesystem marker found at 0x0
[   45.446571] jffs2_build_filesystem(): unlocking the mtd device... 
[   45.446633] done.
[   45.462731] jffs2_build_filesystem(): erasing all blocks after the end marker... 
[   45.747969] rt3050-esw 10110000.esw: link changed 0x01
[   46.218243] br-lan: port 1(eth0.2) entered blocking state
[   46.229013] br-lan: port 1(eth0.2) entered disabled state
[   46.240081] device eth0.2 entered promiscuous mode
[   46.249615] device eth0 entered promiscuous mode
[   46.384903] br-lan: port 1(eth0.2) entered blocking state
[   46.395666] br-lan: port 1(eth0.2) entered forwarding state
[   46.406933] IPv6: ADDRCONF(NETDEV_UP): br-lan: link is not ready
[   47.266820] IPv6: ADDRCONF(NETDEV_CHANGE): br-lan: link becomes ready
[   61.548215] done.
[   61.552070] jffs2: notice: (1123) jffs2_build_xattr_subsystem: complete building xattr subsystem, 0 of xdatum (0 unchecked, 0 orphan) and 0 of xref (0 dead, 0 orphan) found.
[   61.785869] overlayfs: upper fs does not support tmpfile.
[  126.275309] random: crng init done

ynezz commented:
>  [ 0.775110] mtk_soc_eth 10100000.ethernet: generated random MAC address 2a:cb:a9:1e:83:42

I've also [fixed](https://git.openwrt.org/?p=openwrt/staging/ynezz.git;a=blob;f=target/linux/ramips/dts/RAK633.dts;h=ac7218510f7981e0ea530f90559f557ded508b11;hb=39f86f6e36486357cb157997fe3bb721b72c9397#l98) this one, so please [take a look](https://git.openwrt.org/?p=openwrt/staging/ynezz.git;a=blob;f=target/linux/ramips/dts/RAK633.dts;h=ac7218510f7981e0ea530f90559f557ded508b11;hb=39f86f6e36486357cb157997fe3bb721b72c9397#l98) and if you agree with the changes, please close this PR. Thanks.

eikepedia commented:
@ynezz The changes are ok and I agree

adrianschmutzler commented:
Hi, I'm currently reviewing the MAC address assignment on ramips target.
For your device, you are currently using MAC address from <&factory 0x28> and then just calculate WAN address by adding +1 to that.
However, there might be another MAC address stored on flash for your device in <&factory 0x2e>.

Can you check that? If yes, I would also be interested in which of the two addresses should be "lan" and which "wan" then ...
